lib/internal/util/inspect.js

```diff
@@ -1051,7 +1056,49 @@ function getBoxedBase(value, ctx, keys, constructor, tag) {
   return ctx.stylize(base, type.toLowerCase());
 }
 
+function getClassBase(string, value, constructor, tag) {
```

Comment:
this can be done using `class.name` and `Object.getPrototypeOf(class).name`

Reply:
Yes and no: this implementation will print the original name and a changed name in case someone changed either the base class name or the super class name.

I am fine with implementing it differently but there is probably not much benefit doing it that way?

Reply:
what does "changed name" mean?

Reply:
```js
class Foo extends Error {}
Object.defineProperty(Foo, 'name', { value: 'Bla' })
Object.setPrototypeOf(Foo, TypeError.prototype)
```

Should this only be: `class Bla extends TypeError`? It will currently print `class Foo [Bla] [TypeError] extends Error` (I have to fix the prototype positioning. It should be `class Foo [Bla] extends Error [TypeError]`).

Reply:
oh... yeah the `name` property should be the source of truth (otherwise it wouldn't be a configurable property in the first place). i'd be pretty annoyed if node showed classes from my class factory as `class UsefulName [UselessName] {}`

Reply:
Using `class {}` in a factory would result in `[class UsefulName]` after defining a name. I am fine to change it though, since I have no strong opinion on it either way.

Reply:
I'd rather it was changed. (plus getting an anonymous class declaration is surprisingly difficult if you're not familiar with how classes inherit names from declarations)

Reply:
I just updated the code accordingly. PTAL

Reply:
Ping @devsnek